File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/application/ApplicationModule.java
package com.jaynewstrom.concretesample.application;

import android.app.Application;
import android.content.Context;

import dagger.Module;
import dagger.Provides;

@Module
final class ApplicationModule {

    private final Application application;

    ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides @ForApplication Context provideApplicationContext() {
        return application;
    }
}

File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/application/ApplicationModule.kt
package com.jaynewstrom.concretesample.application

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides

@Module
internal class ApplicationModule(private val application: Application) {
    @Provides @ForApplication fun provideApplicationContext(): Context {
        return application
    }
}
